### DISCUSSION TITLE
Is there a way to make vcpkg in manifest mode compile for needed macOS version from project CMakeLists.txt?

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/vcpkg#20151] [Q&A] [answered]
Hi everyone!

I've been recently trying to move from `FetchContent`-based dependency management to vcpkg in the manifest mode for our cross-platform project and in particular, I'm trying to make vcpkg compile the dependencies for macOS 10.14. In CMake, I've been using `CMAKE_OSX_DEPLOYMENT_TARGET` to make it work and I've found out that vcpkg has `VCPKG_OSX_DEPLOYMENT_TARGET`, but there is not much info about it apart from [`triplets.md`](https://github.com/microsoft/vcpkg/blob/master/docs/users/triplets.md). I know that you can set it directly in the triplet file, e.g. in `x64-osx`:
```cmake
...
set(VCPKG_CMAKE_SYSTEM_NAME Darwin)
set(VCPKG_OSX_ARCHITECTURES x86_64)
set(VCPKG_OSX_DEPLOYMENT_TARGET 10.14)
```

It works as expected, but that's not very convenient since I have to patch vcpkg repo, I'd rather set it in the project like this:
```cmake
set(VCPKG_OSX_DEPLOYMENT_TARGET 10.14)
include(${VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake)
```

What I expect in this case is that vcpkg grabs `vcpkg.json` and compiles all the dependencies with fro `10.14`. Unfortunately, `VCPKG_OSX_DEPLOYMENT_TARGET` is ignored in this case. Is there a way to make it work in such a manner?

**Comment by ghost** [comment, marked as answer]
**UPD.** I've just found out about triplet overlays, so now I can store custom triplets in my repo and use them. I guess it resolves the issue with `VCPKG_OSX_DEPLOYMENT_TARGET`.

**Reply by autoantwort** [reply]
Yes this is the way to do it.

**Reply by thomthom** [reply]
This would be very good information to be clarified in the documentation. I was also struggling to figure out the canonical way to control the macOS deployment target of my CMake+Vcpkg project.
